pkg/controller/provider/web/base/auth.go

```diff
@@ -142,11 +144,17 @@ func (r *Auth) permit(token string, p *api.Provider) (allowed bool, err error) {
 		return
 	}
 
-	allowed = review.Status.Allowed
+	if allowed = review.Status.Allowed; !allowed {
+		groupResource := &schema.GroupResource{
+			Resource: resource,
+			Group:    group,
+		}
+		err = fmt.Errorf("%s is forbidden: User %q cannot %s resource %q in API group %q in the namespace %q",
```

Comment:
why %q? isn't %q is for a char? don't you need more than a char?

Reply:
I want this error to match the one that is returned by k8s so I copied it from [here](https://github.com/kubernetes/kubernetes/blob/v1.27.0-alpha.0/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/errors.go#L67)

Reply:
I looked into the integer part :man_facepalming: